packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/CustomVisConfigTabs.tsx

```diff
@@ -76,10 +76,13 @@ const CustomVisConfigTabs: React.FC = memo(() => {
     useEffect(() => {
         if (!isCustomConfig) return;
         const fields = visualizationConfig.chartConfig.fields || [];
-        initVegaLazySchema(fields).then((vegaSchemas) => {
+
+        (async () => {
```

Comment:
nitpick: I'd avoid the short-form/golf here, and implement this as a named function or closure instead for readability:

```
const initVegaAsync = async () => { 
  schemas.current = await initVegaLazySchema(fields);
  setIsLoading(false);
};

void initVegaAsync();
```